### PR TITLE
ci: add release and nix workflows

### PR DESCRIPTION
**Pull request by rvolosatovs**
Refs https://github.com/enarx/enarx/issues/1876 https://github.com/enarx/enarx/issues/1888

Building arm64 binaries on CI in QEMU is too slow, therefore we should add support for cross-compilation to the flake in #1888. I have a draft for that already.
Mac binaries are still non-portable, that should be fixed in #1876.

To build a static `linux-aarch64` package via QEMU: 
```
nix build '.#packages.aarch64-linux.enarx-static'
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+jobs:
+  build:
+    strategy:
+      matrix:
+        platform:
+        - os: ubuntu-20.04
+          target: x86_64-unknown-linux-musl
+        - os: macos-11
+          target: x86_64-apple-darwin
+        # TODO: provide aarch64 builds
```
Comment (rjzak) [inline .github/workflows/release.yml:26]: How about macOS [fat binary](https://developer.apple.com/documentation/apple-silicon/building-a-universal-macos-binary) with the `lipo` tool?

Comment (rvolosatovs) [reply .github/workflows/release.yml:26]: Is it reproducible?

**Comment by bstrie** [comment]
Hm, the nix CI actions appear to be stuck.

**Comment by rvolosatovs** [comment]
> Hm, the nix CI actions appear to be stuck.

Yes, but it does not look like a workflow issue:
```
Waiting for a runner to pick up this job... 
```

I'll try to re-push the HEAD

**Comment by rvolosatovs** [comment]
@bstrie removed MacOS from test matrix, by some reason MacOS `nix` workflows would never start, all green now.

**Review by haraldh — APPROVED** [review]

**Pull request merged**